react-native/debugging/handle-gradle-dependencies-clash.mo.md

```diff
@@ -0,0 +1,116 @@
+# [MO] Handle gradle dependencies versions clashes
+
+## Owner: [Louis Zawadzki](https://github.com/louiszawadzki)
+
+## Prerequisites:
+
+You have 2 dependencies that depend on different versions of the same package, so your application won't build and error with an error like:
+
+```
+com.android.dex.DexException: Multiple dex files define Landroid/support/v7/appcompat/R$anim;
+```
+
+If you're having an issue with Google Maps and Firebase you can have a look at [this article](https://medium.com/@suchydan/how-to-solve-google-play-services-version-collision-in-gradle-dependencies-ef086ae5c75f).
+
+## Steps:
+
+### 1. Know which dependencies is causing the issue *(~2min)*
+
+First, you need to figure out which package is causing the dependency issue.
+
+You can have versions clash without any issue.
+However when one method is present in one version and not in the other there will be an issue.
+
+Look at your error to know exactly which package is causing the issue.
+
+Then you must find out which version should be set.
+
+To do so, run `cd android && ./gradlew app:dependencies && cd ..`.
+This will print out your tree of dependencies.
+
+In there, you must look for two instances of the dependency but with different versions.
+For example if it returns:
+
+```
+_debugCompile - ## Internal use, do not manually configure ##
++--- project :react-native-maps
```

Comment:
Lots of noise in this tree: maybe can you trim some of the dependencies that doesn't add any value to your explanations? Eg. com.google.android.gms:play-services-basement, com.google.android.gms:play-services-tasks etc...

Reply:
I wanted to keep this to show that there can be more dependencies to solve once you've fixed your initial one. I'll shorten it here and then expand it in another section